src/rgw/motr/gc/gc.cc

```diff
@@ -35,18 +35,19 @@ void *MotrGC::GCWorker::entry() {
 
 void MotrGC::initialize() {
   // fetch max gc indices from config
-  auto max_indices = std::min(cct->_conf->rgw_gc_max_objs,
-                              GC_MAX_SHARDS_PRIME);
-  ldpp_dout(this, 50) << __func__ << ": max_indices = " << max_indices << dendl;
-
+  max_indices = static_cast<int>(std::min(cct->_conf->rgw_gc_max_objs,
+                              GC_MAX_SHARDS_PRIME));
```

Comment:
Need to change this name & its value
const int32_t GC_DEFAULT_QUEUES = 1024;

Reply:
`GC_MAX_SHARDS_PRIME` is just a boundary to restrict extremely high values set by the user during configuration.
`1024` should value of `rgw_gc_max_objs` in configuration (solution.config) right?

Reply:
This is basically used to restrict an exceptionally large number supplied by the user.

Reply:
Added GC_DEFAULT_QUEUES with value 128

Reply:
sorry what we wanted to do was the following I guess, please confirm @sachinpunadikar 
```c++
// file: src/common/options/rgw.yaml.in
- name: motr_max_gc_indexes
  type: int
  level: advanced
  desc: Number of shards for garbage collector data
  default: 32
  services: 
  - rgw
```

```c++
// file: gc.h
GC_MAX_INDEXES = 4096 
```

```c++
//file gc.cc
max_indices = static_cast<int>(std::min(cct->_conf->motr_max_gc_indexes,
                                        GC_MAX_INDEXES));
```

Reply:
The current default is 32, but can be fine tuned later.
Else I suppose 64 will be best bet as of now & we will revisit this later.
I agree with GC_MAX_QUEUES (lets call it queues only)
So eventually we need to validate that the size is in between GC_DEFAULT_QUEUES  & GX_MAX_QUEUES. If the set value is less than default, then set it to default & if greater than max, then set it to max

Reply:
Hi, why should we restrict the lower boundary to DEFAULT_QUEUES?? 
for a small setup of say 3 nodes, should not we allow the user to set it to 9 or so?